Add SIMILARNAME warning when variables have names that only differ in lexical case (#7992) (#8020)
verilator/verilator: docs/guide/warnings.rst

   simulators.

.. option:: SIMILARNAME

>    Warning that two or more  variable names differ only in their lexical case.

Reviewer: ```suggestion
   Warns that a variable name only differs from another in lexical case.
```

   Faulty example:

   .. code-block:: sv
      :linenos:

      reg Abcd;

>       wire abcd;

Reviewer: Please use :include for the example, and also an include to show the resulting warning.

Author: done in the latest checkin

   This is still valid Verilog and will not effect Verilator simulation. However
   some downstream VLSI tools may not distinguish net and gate names with the

>    same case and may result in damaged netlists and hard to debug errors.

Reviewer: ```suggestion
   Disabled by default as this is a code-style warning; it will simulate
   correctly.

   This is a warning as some downstream VLSI tools do
   not distinguish net and gate names with the same case.
```
